clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerMembershipManager.java

```diff
@@ -158,10 +157,15 @@ public ConsumerMembershipManager(String groupId,
             commitRequestManager,
             metadata,
             logContext,
-            clientTelemetryReporter,
             backgroundEventHandler,
             time,
             new ConsumerRebalanceMetricsManager(metrics));
+
+        // Update the group member ID label in the client telemetry reporter.
```

Comment:
Could you please remove `clientTelemetryReporter` from this class? `ConsumerMembershipManager` does not use it actually. We can update `clientTelemetryReporter` after creating `ConsumerMembershipManager`. for example:
```java
                    membershipManager = new ConsumerMembershipManager(
                            groupRebalanceConfig.groupId,
                            groupRebalanceConfig.groupInstanceId,
                            groupRebalanceConfig.rebalanceTimeoutMs,
                            serverAssignor,
                            subscriptions,
                            commitRequestManager,
                            metadata,
                            logContext,
                            clientTelemetryReporter,
                            backgroundEventHandler,
                            time,
                            metrics);
                    // Update the group member ID label in the client telemetry reporter.
                    // According to KIP-1082, the consumer will generate the member ID as the incarnation ID of the process.
                    // Therefore, we can update the group member ID during initialization.
                    if (clientTelemetryReporter.isPresent())
                        clientTelemetryReporter.get().updateMetricsLabels(Map.of(ClientTelemetryProvider.GROUP_MEMBER_ID,
                            membershipManager.memberId()));
```

Reply:
I also apply this change to `shareMembershipManager`